jwst/photom/tests/test_photom.py

```diff
@@ -0,0 +1,1470 @@
+        An open data model for a NIRISS image photom reference file.
+    """
+
+    # The middle row should be selected.
+    filter = ["F430M", "CLEAR", "CLEAR"]
+    pupil = ["F090W", "F140M", "F140W"]
+    order = [0, 0, 1]
+
+    nrows = len(filter)
+    nx = 3
+
+    # [3.1, 3.2, 3.3]
+    photmjsr = np.linspace(3.1, 3.1 + (nrows - 1.) * 0.1, nrows)
+    uncertainty = np.zeros(nrows, np.float32)
+    nelem = np.zeros(nrows, np.int32)           # must be zero for image data
+    wavelength = np.ones((nrows, nx), np.float32)
+    relresponse = np.zeros((nrows, nx), np.float32)
+
+    dtype = np.dtype([('filter', 'S12'),
+                      ('pupil', 'S12'),
+                      ('order', '<i2'),
+                      ('photmjsr', '<f4'),
+                      ('uncertainty', '<f4'),
+                      ('nelem', '<i2'),
+                      ('wavelength', '<f4', (nx,)),
+                      ('relresponse', '<f4', (nx,))])
+    reftab = np.array(list(zip(filter, pupil, order,
+                               photmjsr, uncertainty, nelem,
+                               wavelength, relresponse)),
+                      dtype=dtype)
+    ftab = datamodels.NirissPhotomModel(phot_table=reftab)
+
+    return ftab
+
+
+def create_photom_miri_mrs(shape, value, pixel_area, photmjsr):
+    """Create a photom reference file for MIRI MRS.
+
+    Parameters
+    ----------
+    shape : tuple
+        The shape to use when creating image arrays.
+
+    value : float
+        The value to assign to the SCI data array.
+
+    pixel_area : float
+        The pixel solid angle in steradians.
+
+    photmjsr : float
+        The value to assign to the MJy / sr keyword.
+
+    Returns
+    -------
+    ftab : `~jwst.datamodels.DataModel`
+        An open data model for a MIRI MRS photom reference file.
+    """
+
+    data = np.zeros(shape, dtype=np.float32) + value
+    err = np.ones(shape, dtype=np.float32)
+    dq = np.zeros(shape, dtype=np.uint32)
+    pixsiz = np.zeros(shape, dtype=np.float32) + pixel_area
+
+    ftab = datamodels.MiriMrsPhotomModel(data=data, err=err, dq=dq,
+                                         pixsiz=pixsiz)
+    ftab.meta.photometry.conversion_megajanskys = photmjsr
+
+    return ftab
+
+
+def create_photom_miri_lrs(min_wl=5.0, max_wl=10.0, min_r=8.0, max_r=9.0):
+    """Create a photom table for MIRI LRS.
+
+    Parameters
+    ----------
+    min_wl : float
+        Minimum wavelength to assign when populating an array of wavelengths.
+
+    max_wl : float
+        Maximum wavelength to assign when populating an array of wavelengths.
+
+    min_r : float
+        Minimum value to assign when populating the relresponse array.
+
+    max_r : float
+        Maximum value to assign when populating the relresponse array.
+
+    Returns
+    -------
+    ftab : `~jwst.datamodels.DataModel`
+        An open data model for a MIRI LRS photom reference file.
+    """
+
+    filter = ["F560W", "F770W", "F1000W"]
```

Comment:
If this routine is being used to create a photom table for the LRS mode, then the filter value should be "P750L", which is the MIRI prism location in the filter wheel. The ones listed here are imaging filters.

Reply:
Thanks!  I fixed this, here and in line 1299.